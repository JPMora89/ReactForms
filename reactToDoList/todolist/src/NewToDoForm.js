import React, {useState} from 'react';
import { v4 as uuid } from 'uuid';

const NewToDoForm = ({addTodo}) => {
    const INITIAL_STATE = {task: ""};
    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = evt => {
        const {name, value} = evt.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }));
    };

    const handleSubmit = evt => {
        evt.preventDefault();
        const newTodo = {...formData, id: uuid()};
        addTodo(newTodo);
        setFormData(INITIAL_STATE);
    };

    return (
        <div>
            <form onSubmit={handleSubmit}>
                <label htmlFor="task">Task:</label>
                <input

                    type="text"
                    name="task"
                    id="task"
                    placeholder='Task'
                    value={formData.task}
                    onChange={handleChange} />

                <button>Add a new task!</button>
            </form>
        </div>
    );


}

export default NewToDoForm;