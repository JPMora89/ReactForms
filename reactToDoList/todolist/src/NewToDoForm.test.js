import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewToDoForm from './NewToDoForm';

it('renders without crashing', () => {
    render(<NewToDoForm />);
    });

it('matches snapshot', () => {
    const { asFragment } = render(<NewToDoForm />);
    expect(asFragment()).toMatchSnapshot();
});

it('can add a new todo', () => {
    const { getByLabelText, queryByText } = render(<NewToDoForm />);
    // no todos yet
    expect(queryByText('X')).not.toBeInTheDocument();
    // fill out the form
    const taskInput = getByLabelText('Task:');
    fireEvent.change(taskInput, { target: { value: 'Walk the dog' } });
    // submit form
    const submitBtn = queryByText('Add a new task!');
    fireEvent.click(submitBtn);
    // todo exists!
    const removeBtn = queryByText('X');
    expect(removeBtn).toBeInTheDocument();
    expect(removeBtn.previousSibling).toHaveTextContent('Walk the dog');
}   
);
