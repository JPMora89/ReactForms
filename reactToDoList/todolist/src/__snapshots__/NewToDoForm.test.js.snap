// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="task"
      >
        Task:
      </label>
      <input
        id="task"
        name="task"
        placeholder="Task"
        type="text"
        value=""
      />
      <button>
        Add a new task!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
